crates/core/src/ring/connection_manager.rs

```diff
@@ -297,11 +294,11 @@ impl ConnectionManager {
 
     /// Record the advertised location for a peer that we have decided to accept.
     ///
-    /// Pending peers are tracked separately so that other operations cannot route through them
-    /// until the handshake is fully complete. Once the connection is established the entry is
-    /// removed automatically via `prune_in_transit_connection`.
+    /// This makes the peer discoverable to the routing layer even before the connection
+    /// is fully established. The entry is removed automatically if the handshake fails
+    /// via `prune_in_transit_connection`.
     pub fn record_pending_location(&self, peer_id: &PeerId, location: Location) {
-        let mut locations = self.pending_locations.write();
+        let mut locations = self.location_for_peer.write();
```

Comment:
wait why are we writting a pending location into the definitive map?

Reply:
`location_for_peer` now explicitly holds pending entries only for de-duping in-flight handshakes; routing still keys off `connections_by_location`, so these pending inserts are just to keep promotion bookkeeping consistent and avoid duplicate acceptances.

Reply:
location_for_peer containe also definitive connections too though no? Can we document this throughtfully for future reference?